src/sqlfluff/core/linter/linter.py

```diff
@@ -0,0 +1,760 @@
+                        vs.append(ignore_entry)
+                    elif ignore_entry:
+                        ignore_buff.append(ignore_entry)
+            if ignore_buff:
+                linter_logger.info("Parsed noqa directives from file: %r", ignore_buff)
+
+        if tree:
+            t0 = time.monotonic()
+            linter_logger.info("LINTING (%s)", fname)
+            tree, initial_linting_errors = self.lint_fix(
+                tree,
+                config=config,
+                fix=fix,
+                fname=fname,
+                templated_file=parsed.templated_file,
+            )
+            # Update the timing dict
+            t1 = time.monotonic()
+            time_dict["linting"] = t1 - t0
+
+            # We're only going to return the *initial* errors, rather
+            # than any generated during the fixing cycle.
+            vs += initial_linting_errors
+
+        # We process the ignore config here if appropriate
+        if config:
+            for violation in vs:
+                violation.ignore_if_in(config.get("ignore"))
+
+        linted_file = LintedFile(
+            fname,
+            vs,
+            time_dict,
+            tree,
+            ignore_mask=ignore_buff,
+            templated_file=parsed.templated_file,
+        )
+
+        # This is the main command line output from linting.
+        if self.formatter:
+            self.formatter.dispatch_file_violations(
+                fname, linted_file, only_fixable=fix
+            )
+
+        # Safety flag for unset dialects
+        if config.get("dialect") == "ansi" and linted_file.get_violations(
+            fixable=True if fix else None, types=SQLParseError
+        ):
+            if self.formatter:
+                self.formatter.dispatch_dialect_warning()
+
+        return linted_file
+
+    def paths_from_path(
+        self,
+        path: str,
+        ignore_file_name: str = ".sqlfluffignore",
+        ignore_non_existent_files: bool = False,
+        ignore_files: bool = True,
+        working_path: str = os.getcwd(),
+    ) -> List[str]:
+        """Return a set of sql file paths from a potentially more ambiguous path string.
+
+        Here we also deal with the .sqlfluffignore file if present.
+
+        When a path to a file to be linted is explicitly passed
+        we look for ignore files in all directories that are parents of the file,
+        up to the current directory.
+
+        If the current directory is not a parent of the file we only
+        look for an ignore file in the direct parent of the file.
+
+        """
+        if not os.path.exists(path):
+            if ignore_non_existent_files:
+                return []
+            else:
+                raise IOError("Specified path does not exist")
+
+        # Files referred to exactly are also ignored if
+        # matched, but we warn the users when that happens
+        is_exact_file = not os.path.isdir(path)

```

Comment:
Nitpick: Would it be correct (and clearer) to say:
```
is_exact_file = os.path.isfile(path)
```

Reply:
I _think_ so. Will try it at least 🤔